File: Medium/find_first_and_last_position_of_element_in_sorted_array.go
package main

import (
	"fmt"
)

func searchRange(nums []int, target int) []int {
	collectRange := []int{}
	for i, n := range nums {
		if n == target {
			collectRange = append(collectRange, i)
		}
	}

	if len(collectRange) == 0 {
		return []int{-1, -1}
	} else if len(collectRange) == 1 {
		collectRange = append(collectRange, collectRange[0])
	}

	return []int{collectRange[0], collectRange[len(collectRange)-1]}
}

func run_searchRange() {
	fmt.Println(searchRange([]int{5, 7, 7, 8, 8, 10}, 8))
	fmt.Println(searchRange([]int{1}, 1))
	fmt.Println(searchRange([]int{3, 3, 3}, 3))
	fmt.Println(searchRange([]int{0, 0, 1, 2, 2}, 5))
}

File: Medium/find_first_and_last_position_of_element_in_sorted_array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {
    let collectRange = [];
    nums.forEach((n, i) => {
        if(n == target) {
            collectRange.push(i);
        }
    });

    if(collectRange.length == 0) {
        return [-1,-1];
    } else if (collectRange.length == 1) {
        return Array(2).fill(collectRange).flat()
    } 

    return [collectRange[0], collectRange[collectRange.length-1]];
};

console.log(searchRange([5,7,7,8,8,10], 8));
console.log(searchRange([1], 1));
console.log(searchRange([3,3,3], 3));
console.log(searchRange([0,0,1,2,2], 2));